synapse/models/inet.py

```diff
@@ -15,9 +15,43 @@
 
 logger = logging.getLogger(__name__)
 fqdnre = regex.compile(r'^[\w._-]+$', regex.U)
+srv6re = regex.compile(r'^\[([a-f0-9:]+)\]:(\d+)$')
+
 cidrmasks = [((0xffffffff - (2 ** (32 - i) - 1)), (2 ** (32 - i))) for i in range(33)]
 
 
+class Cidr4(s_types.Type):
+
+    def postTypeInit(self):
+        self.setNormFunc(str, self._normPyStr)
+
+    def _normPyStr(self, valu):
+        ip_str, mask_str = valu.split('/', 1)
+
+        mask_int = int(mask_str)
+        if mask_int > 32 or mask_int < 0:
+            raise s_exc.BadTypeValu(valu, mesg='Invalid CIDR Mask')
+
+        ip_int = self.modl.type('inet:ipv4').norm(ip_str)[0]
+
+        mask = cidrmasks[mask_int]
+        network = ip_int & mask[0]
+        broadcast = network + mask[1] - 1
+        network_str = self.modl.type('inet:ipv4').repr(network)
+
+        norm = '{}/{}'.format(network_str, mask_int)
```

Comment:
i thought we banned these .format() calls? ( for being both longer and slower )

Reply:
I though we were going to *only* use these. Had discussion, going to use f-strings.